### PR TITLE
Optimization of function for converting mask images to uint8 type

### PR DESCRIPTION
**Pull request by lcolok**
Dear friend,

Thank you so much for your continuous contributions.

During recent usage, I found some potential improvements when converting mask images to uint8 type. I've adapted the original `(mask*255.0).astype(np.uint8)` operation into a new function: `optimized_mask_to_uint8(mask)`. This function has the following processing logic:

1. Checks whether the input is a numpy array, and if not, an error will be escalated.
2. Adapts the input mask to ensure all values are within the range [0,1]. If any values are out of this range, a prompt will be displayed.
3. If the mask is of floating-point type, it's converted to np.float32 to prevent overflow when being multiplied by 255.
4. Converts the adapted mask to uint8 type.
5. If the mask is 3D and the first dimension is 1, dimensional reduction is performed.

This is my small improvement to the function. I hope we can discuss it together to see if it is suitable for adoption. I really appreciate your past efforts and look forward to further collaboration!

Best regards

### CODE REVIEW

File: modules/impact/core.py [hunk 1]
```diff
@@ -961,8 +979,46 @@ def detect_combined(self, image, threshold, dilation):
...
+            raise ValueError("The type of the input mask needs to be numpy.ndarray!")
+
+        # Clamping, make sure values are within 0-1
+        if np.any(mask < 0) or np.any(mask > 1):
+            print("Warning: Values within the mask are out of the range [0,1], will be clamped.")
+        mask_clamped = np.clip(mask, 0, 1)
+
+        # If mask is floating-point type, convert it to np.float32 to avoid overflow
+        if mask_clamped.dtype.kind == 'f':
+            mask_clamped = mask_clamped.astype(np.float32)
```
Comment (ltdrdata) [inline modules/impact/core.py:1004]: Since it has already been clamped between 0 and 1, is there a need to consider overflow?

File: modules/impact/core.py [hunk 2]
```diff
@@ -814,30 +815,47 @@ def make_sam_mask_segmented(sam_model, segs, image, detection_hint, dilation,
...
+            mask = np.expand_dims(dilate_mask(mask.cpu().numpy(), dilation), axis=-1)
+            mask = np.expand_dims(mask, axis=0)  # Add a batch dimension to the mask at the beginning
+            mask = torch.from_numpy(mask)
+            mask = mask.to(device=mask_working_device)
+        else:
+            # Extract the batch, height, and width
+            height, width, _ = image.shape
+            # Create an empty mask with the shape (N, H, W, 1), where N is the batch size, set to 1
+            mask = torch.zeros(
+                (1, height, width, 1), dtype=torch.float32, device=mask_working_device
```
Comment (ltdrdata) [inline modules/impact/core.py:833]: What is the reason for making the mask 4-dim here? The masks used in ComfyUI are 3-dim masks (b, h, w)

**Review by ltdrdata — CHANGES_REQUESTED** [review]

File: modules/impact/core.py (hunk 2 above)
Comment (ltdrdata) [reply modules/impact/core.py:833]: The review was pending due to my mistake.
